### ISSUE TITLE
error: non-private labels cannot appear between .cfi_startproc / .cfi_endproc pairs

### ISSUE THREAD

**Issue by ForsakenHarmony** [Amanieu/corosensei#23]
GitHub Actions run: https://github.com/vercel/turbo/actions/runs/7961098095/job/21731648141?pr=7409#step:11:1401
Toolchain: `nightly-2024-02-16-x86_64-apple-darwin`

```rs
 error: non-private labels cannot appear between .cfi_startproc / .cfi_endproc pairs
   |
note: instantiated into assembly here
  --> <inline asm>:45:1
   |
45 | _stack_init_trampoline_return:
   | ^

error: previous .cfi_startproc was here
   |
note: instantiated into assembly here
  --> <inline asm>:27:1
   |
27 | .cfi_startproc
   | ^
```

May be related to
https://github.com/llvm/llvm-project/issues/72802
https://github.com/libffi/libffi/issues/807
https://reviews.llvm.org/D155245#4657083

**Comment by Amanieu** [comment]
I think this is an LLVM bug (rust nightly recently upgraded to LLVM 18): https://github.com/llvm/llvm-project/issues/82261

**Comment by eigmax** [comment]
Any possible fix/updates for this?

**Comment by Amanieu** [comment]
This is fixed in LLVM main and is currently being backported to the 18.x branch in https://github.com/llvm/llvm-project/pull/83336. Once that is merged this will need to be cherry-picked into rustc's fork of LLVM. See https://github.com/rust-lang/llvm-project/pull/168.

**Comment by ForsakenHarmony** [comment]
Just needs to be merged into rust now: https://github.com/rust-lang/rust/pull/122559

**Comment by ForsakenHarmony** [comment]
Fixed in the latest rust nightly
